### PR TITLE
Add `member` endpoint

### LINKED ISSUE #784: Allow `mls_group` API users to fetch a KeyPackage using a `KeyPackageRef`

### PR DESCRIPTION
**Pull request by kkohbrok**
This PR adds a function to `mls_group` that allows the consumer to fetch a member's `KeyPackage` given the corresponding `KeyPackageRef`. It also adds some documentation to the book, as well as a small test to the `book_code`.

Fixes #784.

### CODE REVIEW

File: openmls/src/group/mls_group/membership.rs [hunk 1]
```diff
@@ -259,6 +259,20 @@ impl MlsGroup {
             .collect())
     }
 
+    /// Get the [`KeyPackage`] of a member corresponding to the given
+    /// [`KeyPackageRef`]. Returns `None` if no matching [`KeyPackage`] can be
```
Comment (franziskuskiefer) [inline openmls/src/group/mls_group/membership.rs:263]: But it's also returning an error. Either describe what that is, or preferably only return an `Option`.

File: openmls/tests/book_code.rs [hunk 2]
```diff
@@ -478,6 +478,21 @@ fn book_operations(ciphersuite: &'static Ciphersuite, backend: &impl OpenMlsCryp
     assert_eq!(members[1].credential().identity(), b"Bob");
     assert_eq!(members[2].credential().identity(), b"Charlie");
 
+    // Check that the `member` and the `members` function are consistent
+    for member in members {
```
Comment (franziskuskiefer) [inline openmls/tests/book_code.rs:482]: Can you also check that the number of members is what we expect? Otherwise this could just skip members.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, only a few nits.

**Review by raphaelrobert — CHANGES_REQUESTED** [review]
Thanks for tackling this, just some nits.

File: openmls/src/group/mls_group/membership.rs [hunk 3]
```diff
@@ -259,6 +259,20 @@ impl MlsGroup {
...
+    /// found in this group.
+    pub fn member(
+        &self,
+        key_package_ref: &KeyPackageRef,
+    ) -> Result<Option<&KeyPackage>, MlsGroupError> {
```
[the remark this answers is not in the document]
Comment (kkohbrok) [reply openmls/src/group/mls_group/membership.rs:268]: It makes sense in some cases to return both. In my mind, it should give you a `None` if there is no such member and an error if something else went wrong. In this case, the only thing that can go wrong is an `ArchitectureError`, so maybe we can get away in this case to just return a `None` instead, but I disagree that this is always the case.

File: openmls/src/group/mls_group/membership.rs (hunk 1 above)
Comment (kkohbrok) [reply openmls/src/group/mls_group/membership.rs:263]: Good point, that should have been in the docs. I changed the function, though to return `None` when an error is bubbled up. (See my reply [here](https://github.com/openmls/openmls/pull/792#discussion_r804456019).)

**Review by franziskuskiefer — DISMISSED** [review]
Thanks

**Review by raphaelrobert — DISMISSED** [review]
Thanks for making the changes!

**Review by franziskuskiefer — APPROVED** [review]

**Pull request merged**